src/jit/lower.cpp

```diff
@@ -2809,7 +2860,7 @@ GenTree* Lowering::LowerNonvirtPinvokeCall(GenTreeCall* call)
 
     InsertPInvokeCallProlog(call);
 
-    if (call->gtCallType != CT_INDIRECT)
+    if ((comp->opts.eeFlags & CORJIT_FLG_PINVOKE_DIRECT_CALLS) == 0 && call->gtCallType != CT_INDIRECT)
```

Comment:
@jkotas good suggestion. I didn't know that I had to bump the JIT/EE interface version for this change since I was only adding values to enums so I did not even consider this. If I have to bump the version anyway, I think this is the right direction to go.

@sivarv: the flag was added with this change. This flag is checked because there is no need for an indirection when performing direct calls to native methods in CoreRT.